dask/dataframe/tests/test_groupby.py

```diff
@@ -404,6 +407,53 @@ def test_groupby_get_group():
         assert_eq(ddgrouped.a.get_group(2), pdgrouped.a.get_group(2))
 
 
+params = [
+    [1, 2, 3, 4],
+    [
+        "location",
+        "category",
+        "percent",
+        "total",
+        ["location", "category"],
+        ["category", "location"],
+        ["location", "percent"],
+        ["percent", "location"],
+        ["category", "percent"],
+        ["percent", "category"],
+        ["location", "category", "percent"],
+        ["location", "percent", "category"],
+        ["category", "location", "percent"],
+        ["category", "percent", "location"],
+        ["percent", "location", "category"],
+        ["percent", "category", "location"],
+    ],
+    ["percent", "category", "total"],
+    ["first", "last", "all"],
+]
+
+
+@pytest.mark.xfail(
+    sys.version_info[0] == 3
+    and sys.version_info[1] == 6
+    and PANDAS_VERSION == "0.23.4",
+    reason="nlargest() behaves differently with an older version of pandas (0.23.4) on Python 3.6 "
+    "with an unexpected keyword argument 'axis'",
+    strict=False,
+)
+@pytest.mark.parametrize("n, indices, name, keep", list(it.product(*params)))
+def test_dataframe_groupby_nlargest(n, indices, name, keep):
+    data = {}
+    data["location"] = ["A", "A", "A", "B", "C", "C", "D", "D", "D", "D", "D"]
+    data["category"] = [5, 5, 5, 3, 2, 4, 2, 3, 2, 5, 2]
+    data["percent"] = [100, 100, 100, 100, 50, 13, 75, 59, 13, 4, 75]
+    data["total"] = [45, 34, 66, 26, 72, 9, 52, 70, 60, 11, 9]
+    df = pd.DataFrame.from_dict(data)
+    dask_df = dd.from_pandas(df, npartitions=3)
+    expected = dask_df.groupby(indices)[name].nlargest(n=n, keep=keep).compute()
```

Comment:
You can leave off the `compute`. `assert_eq` will do that (and check some things about the task graph too).

Not a big deal, but typically we would call this `result` and the pandas output the `expected`.

Reply:
I am afraid that dropping off `compute()` will again lead us to the issue of `metadata` thingie.
The culprit here is again the weird behavior of `pandas` i.e. returning `RangeIndex` or `MultiIndex`.

So different chunks will have different kinds of indices at run time.

Btw, I checked the tests after removing `compute()`.
It turns out that many of them start failing and the point of failure seems to be in the step:
```
a = ddf._meta.dtype
b = res.dtype
> assert eq_dtypes(a, b)
```

I took the parameters of one of the failing tests, and ran only that separately with a different script. On explicit checking, these 2 turn out to be equal. So I think I don't have a clue as to why this is happening.

Using `compute()` again makes all of them pass though.

PS -- Do you want me to send an updated PR with renaming those variables as `result` and `expected`?
Let me know, thanks!

Reply:
In general, we want `result._meta` to be accurate. This is running into the issue you describe in https://github.com/dask/dask/pull/6627/files#r494596346?

Do you know, is this the same issue as https://github.com/pandas-dev/pandas/issues/34809, which https://github.com/pandas-dev/pandas/pull/34998 is attempting to fix?